src/components/Icon/Icon.tsx

```diff
@@ -8,7 +8,7 @@ type IconProps = {
 	size?: number | string;
 	color?: string;
 	className?: string;
-	onClick?: (icon: Icons) => void;
+	onClick?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
```

Comment:
Some implementations of Icon are based on the previous callback, just add event as second parameter
```suggestion
	onClick?: (icon: Icons, e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
```